src/player.cpp

```diff
@@ -7741,74 +7741,64 @@ bool player::can_wear( const item& it, bool alert ) const
     return true;
 }
 
-bool player::can_wield( const item &it, bool alert ) const
+ret_val<bool> player::can_wield( const item &it ) const
 {
-    if( it.is_two_handed(*this) && ( !has_two_arms() || worn_with_flag("RESTRICT_HANDS") ) ) {
-        if( it.has_flag("ALWAYS_TWOHAND") ) {
-            if( alert ) {
-                add_msg( m_info, _("The %s can't be wielded with only one arm."), it.tname().c_str() );
-            }
-            if( worn_with_flag("RESTRICT_HANDS") ) {
-                add_msg( m_info, _("Something you are wearing hinders the use of both hands.") );
-            }
+    if( it.made_of( LIQUID ) ) {
+        return ret_val<bool>::make_failure( _( "Can't wield spilt liquids." ) );
+    }
+
+    if( it.is_two_handed( *this ) && ( !has_two_arms() || worn_with_flag( "RESTRICT_HANDS" ) ) ) {
+        if( worn_with_flag( "RESTRICT_HANDS" ) ) {
+            return ret_val<bool>::make_failure( _( "Something you are wearing hinders the use of both hands." ) );
+        } else if( it.has_flag( "ALWAYS_TWOHAND" ) ) {
+            return ret_val<bool>::make_failure( _( "The %s can't be wielded with only one arm." ), it.tname().c_str() );
         } else {
-            if( alert ) {
-                add_msg( m_info, _("You are too weak to wield %s with only one arm."),
-                         it.tname().c_str() );
-            }
-            if( worn_with_flag("RESTRICT_HANDS") ) {
-                add_msg( m_info, _("Something you are wearing hinders the use of both hands.") );
-            }
+            return ret_val<bool>::make_failure( _( "You are too weak to wield %s with only one arm." ), it.tname().c_str() );
         }
-        return false;
     }
-    return true;
+
+    return ret_val<bool>::make_success();
 }
 
-bool player::can_unwield( const item& it, bool alert ) const
+ret_val<bool> player::can_unwield( const item& it ) const
 {
     if( it.has_flag( "NO_UNWIELD" ) ) {
-        if( alert ) {
-            add_msg( m_info, _( "You cannot unwield your %s" ), it.tname().c_str() );
-        }
-        return false;
+        return ret_val<bool>::make_failure( _( "You cannot unwield your %s." ), it.tname().c_str() );
     }
 
-    return true;
+    return ret_val<bool>::make_success();
+}
+
+bool player::is_wielding( const item& target ) const
+{
+    return &weapon == &target;
 }
 
 bool player::wield( item& target )
 {
-    if( !can_unwield( weapon ) || !can_wield( target ) ) {
+    if( target.is_null() ) {
+        debugmsg( "Tried to wield an empty item." );
```

Comment:
NPC code assumes wielding a null is fine. There must be one standard: either everyone is allowed to wield nulls or no one is (and everyone has to unwield instead).